File: n117-excel-sheet-column-number.go
package main

import "math"

func titleToNumber(s string) int {
	var sum int

	for i, c := range []rune(s) {
		k := int(c - 'A' + 1)
		exp := float64(len(s) - i - 1)
		sum += k * int(math.Pow(26, exp))
	}

	return sum
}

File: n117-excel-sheet-column-number.js
var titleToNumber = function(s) {
    var sum = 0
    for (var i = 0; i < s.length; i++) {
        var k = s[i].charCodeAt(0) - 'A'.charCodeAt(0) + 1
        var exp = s.length - i - 1
        sum += k * Math.pow(26, exp)
    }
    
    return sum
};
